In [14]:
import scipy.stats as stats
import pandas as pd
from sklearn import preprocessing

def chi_square_test(data):
    return (stats.chi2_contingency(data))

In [23]:
data = pd.read_csv('data_1.csv')
d = preprocessing.normalize(data[['Male_Literate','Female_Literate']])
d = pd.DataFrame(d, columns=['Male_Literate','Female_Literate'])
data[['Male_Literate','Female_Literate']]=d

bihar_data = data[data['State name']=='BIHAR']
karnataka_data = data[data['State name']=='KARNATAKA']
bihar_male = bihar_data['Male_Literate']
bihar_female = bihar_data['Female_Literate']
karnataka_male = karnataka_data['Male_Literate']
karnataka_female = karnataka_data['Female_Literate']

print('Bihar----')
print(chi_square_test([bihar_male,bihar_female]))
print('\nKarnataka----')
print(chi_square_test([karnataka_male,karnataka_female]))

Bihar----
(0.03628574812386675, 1.0, 37, array([[0.82951414, 0.83026401, 0.83492547, 0.83104225, 0.82892507,
        0.82696434, 0.8337368 , 0.83951924, 0.83513482, 0.83802114,
        0.82948554, 0.82607945, 0.82974323, 0.83886367, 0.84394204,
        0.84291566, 0.8378538 , 0.83715329, 0.83538151, 0.83873273,
        0.83643764, 0.83781892, 0.83244781, 0.83940705, 0.83605493,
        0.83732914, 0.83493492, 0.84074495, 0.83365496, 0.83668919,
        0.83735438, 0.83967152, 0.83888522, 0.8379204 , 0.83511528,
        0.82867752, 0.83494057, 0.83357679],
       [0.54552572, 0.54601886, 0.54908445, 0.54653066, 0.54513831,
        0.54384885, 0.54830272, 0.55210551, 0.54922212, 0.5511203 ,
        0.54550691, 0.54326691, 0.54567638, 0.55167438, 0.55501415,
        0.55433915, 0.55101025, 0.55054956, 0.54938436, 0.55158827,
        0.55007892, 0.55098731, 0.54745503, 0.55203173, 0.54982723,
        0.55066521, 0.54909066, 0.5529116 , 0.5482489 , 0.55024435,
        0.55068181, 0.55220566

Null Hypothesis: Male and Female literates are same in the states Bihar and Karnataka.

Inference: From the analysis it is observed that the chi square value for Bihar is 0.036 and for Karnataka is 0.077. Both the p-values are 1 which is greater than the alpha values 0.05 and hence the null hypothesis is true. Male and Female literates are same in both the states.